Feature/hope

gabrielevi10 opened the pull request:
Adiciona movimentação básica do jogador, com animações temporárias para teste;
Adiciona colisão de acordo com o tilemap (paredes e objetos);
Novos assets to Hope.

gabrielevi10 committed:
Add a Player class and background, with camera following the player.
We have a new component class Player, which can moves using arrow keys and are followed by the camera. Add a background image with a camera follower at constuctor of RoomState.

mikaelmello committed:
tilemap: add initial position properties

mikaelmello committed:
map: add getter, fix bug on CanWalk and make logical cell size editable

mikaelmello committed:
state: make current tilemap accessible

mikaelmello committed:
player: use cells for walking and collision

gabrielevi10 committed:
Add simple collision (must be improved).

gabrielevi10 committed:
Add collision in Hope closer to his foot.

gabrielevi10 committed:
Add dnew sprites for hope, according to the direction.

gabrielevi10 committed:
Add walking animation (only forwards and backwards).

gabrielevi10 committed:
Refact on players update

gabrielevi10 committed:
Refact on player's update.

mikaelmello commented on assets/map/wallmap.txt:
```diff
@@ -0,0 +1,15 @@
+8 6 1 
```
Para que serve este arquivo?

gabrielevi10 replied:
pra nada, esqueci de deletar isso

mikaelmello committed:
Remove useless comments from TileMap::CanWalk

mikaelmello committed:
Move Direction enum to Helpers

mikaelmello committed:
Improve code readability on sprite picking logic

Combine moves in a helper function, use only Direction to decide which sprite to open

mikaelmello committed:
Remove random printf

mikaelmello committed:
Change signatures of player's sprite opening functions

mikaelmello committed:
Fix bug when redefining last direction using local variable

gabrielevi10 committed:
Delete useless file.
